From b6d6cc52d3f059298a3b5d4a248fc0177f5d0e08 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:23 -0700
Subject: [PATCH 1647/1851] v4l: xilinx: dprx: Fix warnings for UNINIT

commit 140648650137dfce681e75c91078f07330ef0b99 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the UNINIT type.

The actual warnings are:

For xdprxss_parse_of():
Event var_decl: 	Declaring variable "val" without initializer.
Event uninit_use: 	Using uninitialized value "val".

For xdprxss_enum_mbus_code():
Event var_decl: 	Declaring variable "base" without initializer.
Event uninit_use: 	Using uninitialized value "base".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dprxss.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dprxss.c b/drivers/media/platform/xilinx/xilinx-dprxss.c
index 57de15580327..e907f2ca5f96 100644
--- a/drivers/media/platform/xilinx/xilinx-dprxss.c
+++ b/drivers/media/platform/xilinx/xilinx-dprxss.c
@@ -905,7 +905,7 @@ static int xdprxss_enum_mbus_code(struct v4l2_subdev *sd,
 {
 	struct xdprxss_state *xdprxss = to_xdprxssstate(sd);
 	u32 index = code->index;
-	u32 base;
+	u32 base = 0;
 
 	if (xdprxss->bpc == 8)
 		base = 0;
@@ -1039,7 +1039,7 @@ static const struct v4l2_subdev_ops xdprxss_ops = {
 static int xdprxss_parse_of(struct xdprxss_state *xdprxss)
 {
 	struct device_node *node = xdprxss->dev->of_node;
-	u32 val;
+	u32 val = 0;
 	int ret;
 
 	ret = of_property_read_u32(node, "xlnx,bpc", &xdprxss->bpc);
-- 
2.31.1

